### DISCUSSION TITLE
How to halt rollout of a version which is in staged rollout in Production Track?

### DISCUSSION THREAD

**Discussion by shyvum** [fastlane/fastlane#21991] [Q&A]
I tried to figure this out but did not found anything useful.

We currently use fastlane to update rollout of releases as well as promoting from internal testing track to production track.

I tried setting rollout to 0 to halt release but it didn't help.

**Comment by amanda2369weaver** [comment]
When you set the rollout fraction to 0, it halts the release for new users, but it doesn't stop the app update for users who have already received the update. Here are a few steps you can take to stop the rollout completely:

_**Halt the Rollout**_

First, you need to halt the rollout by setting the rollout fraction to 0. You can do this using the google_play_track_update_binary action in Fastlane:


`google_play_track_update_binary(
  package_name: "com.your.package",
  track: "internal", # or "production"
  version_code: 123, # your version code
  rollout: 0 # set rollout to 0 to halt
)`

_**Deactivate the Release**_

To stop the update for users who have already received it, you need to deactivate the release. This can be done using the Google Play Developer API or the Google Play Console.

With the Google Play Developer API, you can use the deactivate method on the edits.deactivate resource:

`require "google/apis/androidpublisher_v3"

service = Google::Apis::AndroidpublisherV3::AndroidPublisherService.new
service.authorization = Google::Auth.get_application_default([])

package_name = "com.your.package"
edit_id = "your_edit_id"

# Deactivate the release
service.deactivate_edit(package_name, edit_id)`

**Alternatively, you can deactivate the release from the Google Play Console:**

1. Go to the Google Play Console
2. Select your app
3. Go to "Release Management" > "App Releases"
4. Find the release you want to [my-sedgwick.com](https://www.my-sedgwick.com/) deactivate and click on the three-dot menu
5. Select "Deactivate Release"
6. Verify the Rollout Status

_**After halting the rollout and deactivating the release, you can verify the rollout status using the google_play_track_get_rollout action:**_

`rollout = google_play_track_get_rollout(
  package_name: "com.your.package",
  track: "internal", # or "production"
  version_code: 123 # your version code
)

puts rollout # this should be 0`

By following these steps, you should be able to completely stop the rollout of an app update, both for new users and users who have already received the update.
